### PR TITLE
Update CODE_OWNERS.txt

### PR DESCRIPTION
**Pull request by jrose-apple**
On a recent PR, a new contributor wasn't sure who they should request as a reviewer; on another, I complained post-merge about the proper code owner not being notified. (Commit privileges doesn't mean you should be the final reviewer for *any* components, only those where it's clear the code owner trusts you.)

We don't advertise this file very much, and it doesn't fully solve the issue, but I think it's worth updating. This at least covers a few more of the cases that were slipping through the cracks.

### CODE REVIEW

**Comment by jrose-apple** [comment]
Any objections from @akyrtzi, @AnnaZaks, @adrian-prantl, or @moiseev?

For the owners of the "corelibs overlays", did I pick the right person and are you okay with being in this list? @parkera, @das / @mwwa, @mike-ferris-apple / @briancroom?

**Comment by gparker42** [comment]
Is Chris the right "everything else" owner?

**Comment by briancroom** [comment]
I believe @mike-ferris-apple was aiming to reduce his ownership and I should rather be listed as the XCTest code owner.

**Comment by adrian-prantl** [comment]
LGTM from my end.

**Comment by jrose-apple** [comment]
@gparker42 I didn't want to coalesce that into this series of updates. These should be just writing down things that are true in practice, not changing or removing existing ownership (except from Chris's catch-all). @lattner and @tkremenek can work out the appropriate thing to do there.

**Comment by tkremenek** [comment]
@lattner and I can discuss that catch-all separately, but I think it makes sense to be somebody else.  Let's tackle that one separately.  This LGTM.

**Review by tkremenek — APPROVED** [review]

**Comment by AnnaZaks** [comment]
LGTM as well.

**Comment by jrose-apple** [comment]
Updated to include @briancroom – would you prefer your Apple address or another address? (This is what you'd expect people to CC when they email the public lists.)

Still waiting on sign-off from: @akyrtzi, @parkera (or @phausler), and @das.

**Comment by akyrtzi** [comment]
LGTM

**Comment by das** [comment]
lgtm

**Comment by briancroom** [comment]
The Apple address is good. Thanks @jrose-apple!

**Comment by jrose-apple** [comment]
@parkera / @phausler ping

**Comment by CodaFi** [comment]
@jrose-apple In light of [Github's new Code Owners Feature](https://github.com/blog/2392-introducing-code-owners), can we co-opt this pull request to also move to the new format?

**Comment by jrose-apple** [comment]
I'd want to check that with @tkremenek as project lead; unfortunately he's on vacation.

**Comment by tkremenek** [comment]
@CodaFi @jrose-apple Let's make whether or not we use the new code owners feature a separate discussion on swift-dev first, and discuss how the feature should be used.  The real advantage of the new feature is how it can be used to gate pull requests being merged — which is something that should be discussed.  I'm happy to merge this pull request as is.

**Comment by jrose-apple** [comment]
Rebased, but no changes from what I originally had laid out here.

@swift-ci Please smoke test

**Pull request merged**
